### ISSUE TITLE
Rootless container support

### ISSUE THREAD

**Issue by v217** [sagemathinc/cocalc#2170]
See https://github.com/opencontainers/runc/pull/774

**Comment by williamstein** [comment]
Please add (a lot more) description of why you created this issue...

**Comment by v217** [comment]
This might be useful for a student, who wishes to install cocalc on his desktop and doesn't have administrator privileges. If cocalc would run on port say 8000, there's nothing in principle, which absolutely needs root privileges (though I might be wrong/probably I am wrong about this last point.)
Rootless containers seem to have been made possible recently? This means added security.
Sorry for not being able to provide more technical feedback. Thanks!

**Comment by haraldschilly** [comment]
Docker lets you select the port you want to open, so, I guess that's not an issue. I'm wondering if you have setup docker correctly? 

For us to look up anything on that, please tell us which OS, which version of Docker, exactly what command you're running, etc.

**Comment by v217** [comment]
Ubuntu 17.04

    sudo docker run --name=cocalc -d -v ~/cocalc:/projects -p 443:443 sagemathinc/sagemathcloud
and 

    sudo docker start/stop cocalc

I followed the Readme and everything worked great! I didn't configure docker, I just typed

    sudo apt-get install docker.io
I just noticed the possibility of rootless containers, so I thought that certainly would be an improvement.

**Comment by v217** [comment]
And I would like to run:

    docker run --name=cocalc -d -v ~/cocalc:/projects -p 443:443 sagemathinc/sagemathcloud

and

    docker start/stop cocalc
without sudo which seems to be possible?
as mentioned in https://github.com/opencontainers/runc/pull/774

**Comment by haraldschilly** [comment]
Well, I haven't tried it. Did you? Your remark about port numbers is good, maybe just change it to one above 1024?

**Comment by v217** [comment]
No, I couldn't figure out how to change the port number.

**Comment by haraldschilly** [comment]
It's the `-p ...` parameter right there in the command. See https://docs.docker.com/engine/userguide/networking/default_network/dockerlinks/ for some examples.

**Comment by haraldschilly** [comment]
so, `docker run --name=cocalc -d -v ~/cocalc:/projects -p 1443:443 sagemathinc/sagemathcloud` and then https://localhost:1443 would be the first thing I try.

**Comment by williamstein** [comment]
Closing, since @haraldschilly answer really totally answers the question.

**Comment by slel** [comment]
Might be worth adding this to the documentation, faq, or wiki, of 
https://github.com/sagemathinc/cocalc-docker

**Comment by anton-mellit** [comment]
@williamstein It doesn't work straight away. I tried to run this on my computer at the faculty where I don't have privileges. Projects don't start. The problem is that rootless docker setup typically has 65536 available user ids, which means only uids between 1 and 65536 can be used, but smc_compute generates very large uids. I hacked smc_compute to generate uids between 10000 and 10000+32768, but a more permanent solution would be better. Has anyone else tried this?
